### PR TITLE
Issue 16370 recurring event

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/calendar/action/EventFormUtils.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+			String.valueOf(Calendar.SATURDAY),  String.valueOf(Calendar.SUNDAY) };
+
+	/**
+	 *
+	 * @param eventForm
+	 * @return
+	 */
+    public static EventAwareContentletForm setEventDefaults(final EventAwareContentletForm eventForm){
+     /*
+		eventForm.setRecurrenceStartsDate(new Date());
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/calendar/action/EventFormUtils.java]: remove this

File: dotCMS/src/main/java/com/dotmarketing/portlets/calendar/business/EventAPIImpl.java [hunk 2]
```diff
@@ -126,9 +126,8 @@ public Event find(String id, boolean live, User user, boolean respectFrontendRol
 	}
 
 	public Event findbyInode(String inode, User user, boolean respectFrontendRoles) throws DotDataException, DotSecurityException {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/calendar/business/EventAPIImpl.java]: I think this should be CloseDBIfOpened

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/action/EditContentletAction.java [hunk 3]
```diff
@@ -1294,16 +1333,27 @@ else if(perAPI.doesUserHavePermission(contentlet, PermissionAPI.PERMISSION_WRITE
 			Matcher m = p.matcher(interval);
 			boolean b = m.matches();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/action/EditContentletAction.java]: rename "m" and "b" to matcher and matches

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImpl.java [hunk 4]
```diff
@@ -676,6 +678,7 @@ private void handleEventRecurrence(Map<String, Object> contentletFormData, Conte
 				contentlet.setBoolProperty("noRecurrenceEnd", false);
 			}else if(contentletFormData.get("noEndDate")!=null && Boolean.parseBoolean(contentletFormData.get("noEndDate").toString())){
 				contentlet.setDateProperty("recurrenceEnd",null);
+				//contentlet.setProperty("recurrenceEnd", null);
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImpl.java]: remove me

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImpl.java [hunk 5]
```diff
@@ -739,10 +742,9 @@ private void handleEventRecurrence(Map<String, Object> contentletFormData, Conte
 				contentlet.setBoolProperty("recurs",false);
 			}
 
-
 	}
 
-	private void _populateContent(Map<String, Object> contentletFormData,
+	private Contentlet _populateContent(Map<String, Object> contentletFormData,
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImpl.java]: rename the method to be align with the java convension

File: dotCMS/src/main/webapp/html/portlet/ext/calendar/edit_event_js_inc.jsp [hunk 6]
```diff
@@ -37,10 +37,10 @@ function submitfmEvent(form, subcmd)
 		
 	<%
 		boolean found = false;
-		for (Field f : fields) {
-			if (f.isListed()) {
+		for (Field fld : fields) {
```
Comment (jdotcms) [inline dotCMS/src/main/webapp/html/portlet/ext/calendar/edit_event_js_inc.jsp]: rename to field

File: dotCMS/src/main/java/com/dotmarketing/portlets/calendar/action/EventFormUtils.java (hunk 1 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotmarketing/portlets/calendar/action/EventFormUtils.java]: done

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/action/EditContentletAction.java (hunk 3 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/action/EditContentletAction.java]: done

File: dotCMS/src/main/java/com/dotmarketing/portlets/calendar/business/EventAPIImpl.java (hunk 2 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotmarketing/portlets/calendar/business/EventAPIImpl.java]: added

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImpl.java (hunk 4 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImpl.java]: done

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImpl.java (hunk 5 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImpl.java]: changed

File: dotCMS/src/main/webapp/html/portlet/ext/calendar/edit_event_js_inc.jsp (hunk 6 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/webapp/html/portlet/ext/calendar/edit_event_js_inc.jsp]: renamed

**Review by jgambarios — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

**Pull request merged**